### ISSUE TITLE
Agent Update from Bundle to Flat MPKG

### ISSUE THREAD

**Issue by zackrspv** [Ylianst/MeshCentral#2476]
There are two types of mkpg installers:

1. Bundle style (legacy way which Apple deprecated) after MacOS 10.5
2. Flat style xar based(10.5 and later uses this style for mpkg packaging)

Currently, MeshCentral2 creates a Bundle Style MPKG file for installation of the Mesh Agent on MacOS. 

It would be ideal if the Agent could be updated to a flat style installer instead, so that it can be properly leveraged on the more recent versions of MacOS.

Here is more information on understanding the package differences https://derflounder.wordpress.com/2014/06/01/understanding-payload-free-packages/

Here's another resource we found on bundle style to flat style package changes required https://macadmins.psu.edu/files/2019/07/psumac2019-345-Installer-Package-Scripting-Making-your-deployments-easier-one-at-a-time.pdf.

Regards, 

--Phillip

**Comment by Ylianst** [comment]
I agree. We do have to create the installer within the MeshCentral server itself as we change the installer depending on the server and device groups you select. We will need to take a look at that.

**Comment by si458** [comment]
I have done loads of customise feature this past few weeks for the osx mpkg file
the mpkg is now customised with filename, group, install locations, etc...
you can test this all out by using the `master` docker image OR `npm install Ylianst/MeshCentral`
